lib/buildkit/client.rb

```diff
@@ -120,6 +128,34 @@ def request(method, path, data, options = {})
       response.data
     end
 
+    def extract_query_and_headers_from(data)
+      {
+        query: data.delete(:query) || {},
+        headers: data.delete(:headers) || {},
+      }
+    end
+
+    def paginate(method, path, data, options = {})
+      response = []
+      loop do
+        request method, path, data, options
+        link_header = parse_link_header(@last_response.headers[:link])
+        @last_response.data.each { |r| response << r }
```

Comment:
This means that `response` is holding the `data`, rather than a list of responses. `Oktokit` calls `response` `data` instead, which I think we should do here. Using `concat`  is also a likely cleaner approach.